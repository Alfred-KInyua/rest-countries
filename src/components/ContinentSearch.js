import React, { useState, useEffect } from 'react';
import { NavLink, useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { FaAnchor } from 'react-icons/fa';
import Nav from './Access/Nav';
import { fetchCountries } from '../redux/countries';

const ContinentSearch = () => {
  let regionDetails = useSelector((state) => state.entities.countries);

  const { title } = useParams();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCountries(title));
  }, []);

  const [searchText, setYourFilterText] = useState('');
  regionDetails = regionDetails.filter((country) => country.name.common.includes(searchText));
  const handleChange = (e) => {
    setYourFilterText(e.target.value.toLowerCase());
  };

  return (
    <div>
      <Nav title={title} />
      <div className="homs displayFlex flex-column">
        <ul className="displayFlex space-btn home-data">
          <li className="text-right">
            <NavLink
              to="/"
              className="navLink displayFlex justify-center align-center"
            >
              <FaAnchor />
              Back to Home
            </NavLink>
          </li>
          <li className="displayFlex flex-column text-right">
            <h2>{title}</h2>
            <p>
              {regionDetails.length}
              Countries
            </p>
          </li>
        </ul>
      </div>
      <div>
        <div className="search">
          <input
            name="query"
            value={searchText}
            onChange={handleChange}
            placeholder="Search Country"
            className="searchTerm"
          />
        </div>
        <ul className="grid-container">
          {regionDetails.map((country) => (
            <NavLink
              to={`/country/${country.name.common}`}
              key={country.name.common}
              className="navLink"
            >
              <li className="country-list displayFlex space-btn">
                <div className="displayFlex justify-center align-center">
                  <img
                    src={country.flags.svg}
                    className="flag"
                    alt={country.flags.svg}
                  />
                  <h3>{country.name.common}</h3>
                </div>
              </li>
            </NavLink>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default ContinentSearch;