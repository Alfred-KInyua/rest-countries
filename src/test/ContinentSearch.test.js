import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import ContinentSearch from '../components/ContinentSearch';

describe('Testing ContinentSearch component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <ContinentSearch />
        </Router>
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
