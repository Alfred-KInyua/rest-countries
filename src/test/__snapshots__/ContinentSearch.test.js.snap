// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContinentSearch component Testing with snapshot 1`] = `
<div>
  <header
    className="header"
  >
    <nav>
      <ul
        className="menu displayFlex"
      >
        <li
          className="flex-1"
        >
          <a
            aria-current="page"
            className="navLink active"
            href="/"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm160 215.5v6.93c0 5.87-3.32 11.24-8.57 13.86l-15.39 7.7a15.485 15.485 0 0 1-15.53-.97l-18.21-12.14a15.52 15.52 0 0 0-13.5-1.81l-2.65.88c-9.7 3.23-13.66 14.79-7.99 23.3l13.24 19.86c2.87 4.31 7.71 6.9 12.89 6.9h8.21c8.56 0 15.5 6.94 15.5 15.5v11.34c0 3.35-1.09 6.62-3.1 9.3l-18.74 24.98c-1.42 1.9-2.39 4.1-2.83 6.43l-4.3 22.83c-.62 3.29-2.29 6.29-4.76 8.56a159.608 159.608 0 0 0-25 29.16l-13.03 19.55a27.756 27.756 0 0 1-23.09 12.36c-10.51 0-20.12-5.94-24.82-15.34a78.902 78.902 0 0 1-8.33-35.29V367.5c0-8.56-6.94-15.5-15.5-15.5h-25.88c-14.49 0-28.38-5.76-38.63-16a54.659 54.659 0 0 1-16-38.63v-14.06c0-17.19 8.1-33.38 21.85-43.7l27.58-20.69a54.663 54.663 0 0 1 32.78-10.93h.89c8.48 0 16.85 1.97 24.43 5.77l14.72 7.36c3.68 1.84 7.93 2.14 11.83.84l47.31-15.77c6.33-2.11 10.6-8.03 10.6-14.7 0-8.56-6.94-15.5-15.5-15.5h-10.09c-4.11 0-8.05-1.63-10.96-4.54l-6.92-6.92a15.493 15.493 0 0 0-10.96-4.54H199.5c-8.56 0-15.5-6.94-15.5-15.5v-4.4c0-7.11 4.84-13.31 11.74-15.04l14.45-3.61c3.74-.94 7-3.23 9.14-6.44l8.08-12.11c2.87-4.31 7.71-6.9 12.89-6.9h24.21c8.56 0 15.5-6.94 15.5-15.5v-21.7C359.23 71.63 422.86 131.02 441.93 208H423.5c-8.56 0-15.5 6.94-15.5 15.5z"
              />
            </svg>
          </a>
        </li>
        <li
          className="flex-1"
        />
      </ul>
    </nav>
  </header>
  <div
    className="homs displayFlex flex-column"
  >
    <ul
      className="displayFlex space-btn home-data"
    >
      <li
        className="text-right"
      >
        <a
          aria-current="page"
          className="navLink displayFlex justify-center align-center active"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.971 352h32.394C67.172 454.735 181.944 512 288 512c106.229 0 220.853-57.38 242.635-160h32.394c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0l-67.029 67.029c-7.56 7.56-2.206 20.485 8.485 20.485h35.146c-20.29 54.317-84.963 86.588-144.117 94.015V256h52c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-52v-5.47c37.281-13.178 63.995-48.725 64-90.518C384.005 43.772 341.605.738 289.37.01 235.723-.739 192 42.525 192 96c0 41.798 26.716 77.35 64 90.53V192h-52c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v190.015c-58.936-7.399-123.82-39.679-144.117-94.015h35.146c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0L4.485 331.515C-3.074 339.074 2.28 352 12.971 352zM288 64c17.645 0 32 14.355 32 32s-14.355 32-32 32-32-14.355-32-32 14.355-32 32-32z"
            />
          </svg>
          Back to Home
        </a>
      </li>
      <li
        className="displayFlex flex-column text-right"
      >
        <h2 />
        <p>
          0
          Countries
        </p>
      </li>
    </ul>
  </div>
  <div>
    <div
      className="search"
    >
      <input
        className="searchTerm"
        name="query"
        onChange={[Function]}
        placeholder="Search Country"
        value=""
      />
    </div>
    <ul
      className="grid-container"
    />
  </div>
</div>
`;
